BlueDBOnDisk/src/main/java/org/bluedb/disk/segment/SegmentBatch.java

```diff
@@ -57,7 +59,7 @@
 
 	protected static Range getLargestEmptyRange(List<Range> optionsSmallToBig, Set<Range> existingChunkRanges) {
 		return optionsSmallToBig.stream()
-				.sorted(Comparator.reverseOrder())
+				.sorted(reverseOrderBySize)

```

Comment:
You should filter before you sort. Actually, you should use the min/max operation instead of findFirst again